pkg/datapath/linux/node.go

```diff
@@ -568,7 +568,7 @@ func (n *linuxNodeHandler) insertNeighbor(newNode *nodeTypes.Node, ifaceName str
 	routes, err := netlink.RouteGet(nextHopIPv4)
 	if err != nil {
 		scopedLog.WithError(err).Error("Failed to retrieve route for remote node IP")
```

Comment:
Just a small note that I forgot in my previous review. Since we are now returning errors, wouldn't it make sense to log the errors in the caller instead of potentially duplicating the log messages? i.e., one would be printed in this line and the other would again printed in the caller.

Reply:
Good point, will fix.